mobile/src/main/java/ch/epfl/sdp/mobile/state/game/ActualPuzzleGameScreenState.kt

```diff
@@ -0,0 +1,105 @@
+import ch.epfl.sdp.mobile.ui.game.*
+import ch.epfl.sdp.mobile.ui.puzzles.PuzzleGameScreenState
+import ch.epfl.sdp.mobile.ui.puzzles.PuzzleInfoState
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.PermissionState
+import kotlinx.coroutines.CoroutineScope
+
+/**
+ * Creates a new [ActualPuzzleGameScreenState].
+ *
+ * @param currentActions the [State] of actions to perform when navigating.
+ * @param currentUser the currently authenticated user.
+ * @param puzzle the [Puzzle] to display.
+ * @param permission the [PermissionState] for the microphone permission.
+ * @param speechFacade the [SpeechFacade] to acecess speech recognition facilities.
+ * @param snackbarHostState the [SnackbarHostState] to display some info.
+ * @param scope a [CoroutineScope] keeping track of the state lifecycle.
+ */
+fun ActualPuzzleGameScreenState(
+    currentActions: State<StatefulGameScreenActions>,
+    currentUser: State<AuthenticatedUser>,
+    puzzle: Puzzle,
+    permission: PermissionState,
+    speechFacade: SpeechFacade,
+    snackbarHostState: SnackbarHostState,
+    scope: CoroutineScope,
+): ActualPuzzleGameScreenState {
+  val puzzleInfo = DelegatingPuzzleInfoState(puzzle)
+
+  // The backing MutableGameDelegate.
+  val chessBoard = PuzzleGameDelegate(currentUser, puzzle, puzzleInfo, scope)
+
+  // The delegates which use the MutableGameDelegate.
+  val promotions = DelegatingPromotionState(chessBoard)
+  val moves = DelegatingMovesInfoState(chessBoard)
+  val moveableChessBoard =
+      DelegatingPuzzlePromotionMovableChessBoardState(
+          currentUser.value,
+          chessBoard,
+          puzzleInfo,
+          promotions,
+      )
+  val speechRecognizer =
+      DelegatingSpeechRecognizerState(
+          chessBoard,
+          permission,
+          speechFacade,
+          snackbarHostState,
+          scope,
+      )
+  return ActualPuzzleGameScreenState(
+      currentActions = currentActions,
+      puzzle = puzzle,
+      moveableChessBoard = moveableChessBoard,
+      promotionState = promotions,
+      movesInfo = moves,
+      puzzleInfo = puzzleInfo,
+      speechRecognizer = speechRecognizer,
+  )
+}
+
+/**
+ * An implementation of [PuzzleGameScreenState] which delegates its behaviors to different
+ * underlying states.
+ *
+ * @param currentActions the [State] of actions to perform when navigating.
+ * @param puzzle the [Puzzle] for this screen.
+ * @param moveableChessBoard the underlying [MovableChessBoardState].
+ * @param promotionState the underlying [PromotionState].
+ * @param movesInfo the underlying [MovesInfoState].
+ * @param puzzleInfo the underlying [PuzzleInfoState].
+ * @param speechRecognizer the underlying [SpeechRecognizerState].
+ */
+class ActualPuzzleGameScreenState
+constructor(
+    private val currentActions: State<StatefulGameScreenActions>,
+    private val puzzle: Puzzle,
```

Comment:
This parameter is not used.

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/309/commits/8ed41c6965f20a742cca43d4005de02f2076382a